internal/api/v1/user/transactions/transactions_test.go

```diff
@@ -0,0 +1,505 @@
+package transactions_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	client "github.com/bitcoin-sv/spv-wallet-go-client"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/querybuilders"
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/testfixtures"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTransactionsAPI_UpdateTransactionMetadata(t *testing.T) {
+	ID := "1024"
+	tests := map[string]struct {
+		code             int
+		responder        httpmock.Responder
+		statusCode       int
+		expectedResponse *response.Transaction
+		expectedErr      error
+	}{
+		fmt.Sprintf("HTTP PATCH /api/v1/transactions/%s response: 200", ID): {
+			expectedResponse: &response.Transaction{
+				Model: response.Model{
+					CreatedAt: testfixtures.ParseTime("2024-10-07T14:03:26.736816Z"),
+					UpdatedAt: testfixtures.ParseTime("2024-10-07T14:03:26.736816Z"),
+					Metadata: map[string]any{
+						"domain":          "john.doe.test.4chain.space",
+						"example_key1":    "example_key10_val",
+						"example_key2":    "example_key20_val",
+						"ip_address":      "127.0.0.01",
+						"user_agent":      "node-fetch",
+						"paymail_request": "HandleReceivedP2pTransaction",
+						"reference_id":    "1c2dcc61-f48f-44f2-aba2-9a759a514d49",
+						"p2p_tx_metadata": map[string]any{
+							"pubkey": "3fa7af5b-4568-4873-86da-0aa442ca91dd",
+							"sender": "john.doe@handcash.io",
+						},
+					},
+				},
+				ID:                   "2c250e21-c33a-41e3-a4e3-77c68b03244e",
+				Hex:                  "283b1c6deb6d6263b3cec7a4701d46d3",
+				XpubOutIDs:           []string{"4c9a0a0d-ea4f-4f03-b740-84438b3d210d"},
+				BlockHash:            "47758f612c6bf5b454bcd642fe8031f6",
+				BlockHeight:          512,
+				Fee:                  1,
+				NumberOfInputs:       2,
+				NumberOfOutputs:      3,
+				TotalValue:           311,
+				OutputValue:          100,
+				Status:               "MINED",
+				TransactionDirection: "incoming",
+			},
+			code:      http.StatusOK,
+			responder: httpmock.NewJsonResponderOrPanic(http.StatusOK, httpmock.File("transactionstest/transaction_update_metadata_200.json")),
```

Comment:
Maybe we could move these transactions or whole folder `transactionstest` to `testfixtures`? WDYT?